### DISCUSSION TITLE
How to group updates by file?

### DISCUSSION THREAD

**Discussion by justyns** [renovatebot/renovate#33053] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket, helm, renovate version 38.83.0

### Please tell us more about your question or problem

I am trying to figure out how to group updates by file automatically.  I have a config like this that is working:

```json
"packageRules": [
  {
    "matchDatasources": ["docker"],
    "matchedFileNames": ["namespaces/dev1/config.yaml"],
    "groupName": "dev1 dependencies",
  },
  {
    "matchDatasources": ["docker"],
    "matchedFileNames": ["namespaces/dev2/config.yaml"],
    "groupName": "dev2 dependencies",
  },
  {
    "matchDatasources": ["docker"],
    "matchedFileNames": ["namespaces/qa1/config.yaml"],
    "groupName": "qa1 dependencies",
  },
]
```

However, I don't want to have to specify every config.yaml in a separate rule.  Is there a way to define one rule that uses regex or a wildcard to match the file and then include part of that file path in the group name?

Or is there a better way in general to handle something like this?  Essentially I am wanting to update all dependencies in a file(namespace) in one PR, instead of one PR for each dependency.  Each of these config.yaml files contain the same or similar dependencies, but we can only deploy updates to a single namespace at a time - e.g. we don't want to deploy an updated image to dev, qa, and prod all at the same time.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
You can use template fields for this. Checkout https://docs.renovatebot.com/templates/#template-fields

**Reply by justyns** [reply]
Thanks, that was easier than expected!  I was able to  use `{{parentDir}}` to get what I needed for this case
